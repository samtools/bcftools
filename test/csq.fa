>1
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
>2
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
>3
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT
ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT

