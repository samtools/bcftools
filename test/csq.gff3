##gff-version   3
#! This file shows which fields are used and required by `bcftools +csq`. It is a trimmed version 
#! of the GFF3 format, see an example of the full format here
#!      ftp://ftp.ensembl.org/pub/grch37/release-84/gff3/homo_sapiens/
#!
###
1	.	gene	90	200	.	+	.	ID=gene:ENSG00000000001;Name=XYZ;biotype=protein_coding
1	.	transcript	90	200	.	+	.	ID=transcript:ENST00000000001;Parent=gene:ENSG00000000001;biotype=protein_coding
1	.	exon	90	110	.	+	.	Parent=transcript:ENST00000000001
1	.	five_prime_UTR	90	98	.	+	.	Parent=transcript:ENST00000000001
1	.	CDS	99	110	.	+	1	Parent=transcript:ENST00000000001
1	.	exon	120	130	.	+	.	Parent=transcript:ENST00000000001
1	.	CDS	120	130	.	+	1	Parent=transcript:ENST00000000001
1	.	exon	140	150	.	+	.	Parent=transcript:ENST00000000001
1	.	CDS	140	150	.	+	2	Parent=transcript:ENST00000000001
1	.	exon	160	200	.	+	.	Parent=transcript:ENST00000000001
1	.	CDS	160	171	.	+	0	Parent=transcript:ENST00000000001
1	.	three_prime_UTR	172	200	.	+	.	Parent=transcript:ENST00000000001
2	.	gene	80	200	.	-	.	ID=gene:ENSG00000000002;Name=ABC;biotype=protein_coding
2	.	transcript	80	200	.	-	.	ID=transcript:ENST00000000002;Parent=gene:ENSG00000000002;biotype=protein_coding
2	.	exon	80	110	.	-	.	Parent=transcript:ENST00000000002
2	.	three_prime_UTR	80	98	.	-	.	Parent=transcript:ENST00000000002
2	.	CDS	99	110	.	-	0	Parent=transcript:ENST00000000002
2	.	exon	120	130	.	-	.	Parent=transcript:ENST00000000002
2	.	CDS	120	130	.	-	2	Parent=transcript:ENST00000000002
2	.	exon	140	150	.	-	.	Parent=transcript:ENST00000000002
2	.	CDS	140	150	.	-	1	Parent=transcript:ENST00000000002
2	.	exon	160	200	.	-	.	Parent=transcript:ENST00000000002
2	.	CDS	160	171	.	-	1	Parent=transcript:ENST00000000002
2	.	five_prime_UTR	172	200	.	-	.	Parent=transcript:ENST00000000002
3	.	lincRNA_gene	20	50	.	-	.	ID=gene:ENSG00000000004;Name=mir-007;biotype=lincRNA
3	.	lincRNA	20	50	.	-	.	ID=transcript:ENSG00000000004;Parent=gene:ENSG00000000004;biotype=lincRNA
3	.	gene	100	200	.	-	.	ID=gene:ENSG00000000003;Name=QWRTY;biotype=protein_coding
3	.	transcript	100	200	.	-	.	ID=transcript:ENST00000000003;Parent=gene:ENSG00000000003;biotype=protein_coding
3	.	exon	100	110	.	-	.	Parent=transcript:ENST00000000003
3	.	three_prime_UTR	100	105	.	-	.	Parent=transcript:ENST00000000003
3	.	CDS	106	110	.	-	0	Parent=transcript:ENST00000000003
3	.	exon	160	200	.	-	.	Parent=transcript:ENST00000000003
3	.	CDS	160	171	.	-	0	Parent=transcript:ENST00000000003
3	.	five_prime_UTR	172	200	.	-	.	Parent=transcript:ENST00000000003
