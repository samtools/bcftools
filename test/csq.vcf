##fileformat=VCFv4.2
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##INFO=<ID=EXP,Number=1,Type=String,Description="Expected consequence">
##contig=<ID=1,length=2147483647>
##contig=<ID=2,length=2147483647>
##contig=<ID=3,length=2147483647>
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	SmplAAA	SmplBBB
1	90	.	C	T	.	.	EXP=5_prime_utr|XYZ|ENST00000000001|+	GT	1|0	1|1
1	102	.	C	T	.	.	EXP=synonymous|XYZ|ENST00000000001|+|1Y>1Y|102C>T	GT	1|0	1|0
1	103	.	G	A	.	.	EXP=missense|XYZ|ENST00000000001|+|2V>2I|103G>A	GT	1|0	0|0
1	103	.	G	C	.	.	EXP=missense|XYZ|ENST00000000001|+|2V>2L|103G>C	GT	0|0	1|0
1	107	.	G	A	.	.	EXP=missense&splice_region|XYZ|ENST00000000001|+|3R>3Q|107G>A+108T>A	GT	1|0	1|0
1	108	.	T	A	.	.	EXP=@107	GT	1|0	1|0
1	121	.	ACG	A	.	.	EXP=inframe_deletion|XYZ|ENST00000000001|+|5TY>5I|121ACG>A+124TA>T	GT	1|0	1|0
1	124	.	TA	T	.	.	EXP=@121	GT	1|0	1|0
1	128	.	T	C	.	.	EXP=missense&splice_region|XYZ|ENST00000000001|+|7V>6A|128T>C+129A>C	GT	1|0	0/0
1	129	.	A	C	.	.	EXP=@128	GT	1|0	0/0
1	140	.	TA	AACG	.	.	EXP=inframe_insertion&splice_region|XYZ|ENST00000000001|+|8LR>7QRR|140TA>AACG+142C>CC	GT	1|0	0|0
1	142	.	C	CC	.	.	EXP=@140	GT	1|0	0|0
1	145	.	AC	TA	.	.	EXP=stop_gained|XYZ|ENST00000000001|+|10T>10*|145AC>TA	GT	1|0	0|0
1	160	.	TA	T	.	.	EXP=*frameshift&splice_region|XYZ|ENST00000000001|+|12YVRT>12SYV|160TA>T	GT	1|0	0|0
1	190	.	C	T	.	.	EXP=3_prime_utr|XYZ|ENST00000000001|+	GT	1|0	0|0
2	97	.	A	C	.	.	EXP=3_prime_utr|ABC|ENST00000000002|-	GT	1|0	0|0
2	105	.	AC	A	.	.	EXP=@121	GT	1|0	0|0
2	121	.	AC	A	.	.	EXP=frameshift|ABC|ENST00000000002|-|11VVRTY>11*|105AC>A+121AC>A	GT	1|0	0|0
2	126	.	C	CTT	.	.	EXP=@127	GT	1|0	0|0
2	127	.	G	GG	.	.	EXP=inframe_insertion|ABC|ENST00000000002|-|9T>8TK|126C>CTT+127G>GG	GT	1|0	0|0
2	144	.	TAC	T	.	.	EXP=@148	GT	1|0	0|0
2	148	.	TA	T	.	.	EXP=inframe_deletion&splice_region|ABC|ENST00000000002|-|5YV>5T|144TAC>T+148TA>T	GT	1|0	0|0
2	164	.	T	G	.	.	EXP=missense|ABC|ENST00000000002|-|3T>3P|164T>G	GT	1|0	0|0
2	165	.	A	C	.	.	EXP=synonymous|ABC|ENST00000000002|-|2R>2R|165A>C	GT	1|0	0|0
2	169	.	A	G	.	.	EXP=@170	GT	1|0	0|0
2	170	.	C	T	.	.	EXP=missense&start_lost|ABC|ENST00000000002|-|1V>1T|169A>G+170C>T	GT	1|0	0|0
2	199	.	G	T	.	.	EXP=5_prime_utr|ABC|ENST00000000002|-	GT	1|0	0|0
3	20	.	T	A	.	.	EXP=non_coding|mir-007|lincRNA	GT	1|0	0|0
3	109	.	ACGTACGT	A	.	.	EXP=splice_acceptor|QWRTY|ENST00000000003|-	GT	1|0	0|0
3	113	.	A	T	.	.	EXP=splice_region|QWRTY|ENST00000000003|-	GT	1|0	0|0
3	120	.	T	A	.	.	EXP=intron|QWRTY	GT	1|0	0|0
3	152	.	T	A	.	.	EXP=splice_region|QWRTY|ENST00000000003|-	GT	1|0	0|0
3	159	.	G	A	.	.	EXP=splice_donor|QWRTY|ENST00000000003|-	GT	1|0	0|0
