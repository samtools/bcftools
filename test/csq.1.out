1	90	.	C	T	.	.	EXP=5_prime_utr|XYZ|ENST00000000001|+;BCSQ=5_prime_utr|XYZ|ENST00000000001|+	GT:BCSQ	1|0:1	1|1:3
1	102	.	C	T	.	.	EXP=synonymous|XYZ|ENST00000000001|+|1Y>1Y|102C>T;BCSQ=synonymous|XYZ|ENST00000000001|+|1Y>1Y|102C>T	GT:BCSQ	1|0:1	1|0:1
1	103	.	G	A	.	.	EXP=missense|XYZ|ENST00000000001|+|2V>2I|103G>A;BCSQ=missense|XYZ|ENST00000000001|+|2V>2I|103G>A	GT:BCSQ	1|0:1	0|0:0
1	103	.	G	C	.	.	EXP=missense|XYZ|ENST00000000001|+|2V>2L|103G>C;BCSQ=missense|XYZ|ENST00000000001|+|2V>2L|103G>C	GT:BCSQ	0|0:0	1|0:1
1	107	.	G	A	.	.	EXP=missense&splice_region|XYZ|ENST00000000001|+|3R>3Q|107G>A+108T>A;BCSQ=missense&splice_region|XYZ|ENST00000000001|+|3R>3Q|107G>A+108T>A	GT:BCSQ	1|0:1	1|0:1
1	108	.	T	A	.	.	EXP=@107;BCSQ=@107	GT:BCSQ	1|0:1	1|0:1
1	121	.	ACG	A	.	.	EXP=inframe_deletion|XYZ|ENST00000000001|+|5TY>5I|121ACG>A+124TA>T;BCSQ=inframe_deletion|XYZ|ENST00000000001|+|5TY>5I|121ACG>A+124TA>T	GT:BCSQ	1|0:1	1|0:1
1	124	.	TA	T	.	.	EXP=@121;BCSQ=@121	GT:BCSQ	1|0:1	1|0:1
1	128	.	T	C	.	.	EXP=missense&splice_region|XYZ|ENST00000000001|+|7V>6A|128T>C+129A>C;BCSQ=missense&splice_region|XYZ|ENST00000000001|+|7V>6A|128T>C+129A>C	GT:BCSQ	1|0:1	0/0:0
1	129	.	A	C	.	.	EXP=@128;BCSQ=@128	GT:BCSQ	1|0:1	0/0:0
1	140	.	TA	AACG	.	.	EXP=inframe_insertion&splice_region|XYZ|ENST00000000001|+|8LR>7QRR|140TA>AACG+142C>CC;BCSQ=inframe_insertion&splice_region|XYZ|ENST00000000001|+|8LR>7QRR|140TA>AACG+142C>CC	GT:BCSQ	1|0:1	0|0:0
1	142	.	C	CC	.	.	EXP=@140;BCSQ=@140	GT:BCSQ	1|0:1	0|0:0
1	145	.	AC	TA	.	.	EXP=stop_gained|XYZ|ENST00000000001|+|10T>10*|145AC>TA;BCSQ=stop_gained|XYZ|ENST00000000001|+|10T>10*|145AC>TA	GT:BCSQ	1|0:1	0|0:0
1	160	.	TA	T	.	.	EXP=*frameshift&splice_region|XYZ|ENST00000000001|+|12YVRT>12SYV|160TA>T;BCSQ=*frameshift&splice_region|XYZ|ENST00000000001|+|12YVRT>12SYV|160TA>T	GT:BCSQ	1|0:1	0|0:0
1	190	.	C	T	.	.	EXP=3_prime_utr|XYZ|ENST00000000001|+;BCSQ=3_prime_utr|XYZ|ENST00000000001|+	GT:BCSQ	1|0:1	0|0:0
2	97	.	A	C	.	.	EXP=3_prime_utr|ABC|ENST00000000002|-;BCSQ=3_prime_utr|ABC|ENST00000000002|-	GT:BCSQ	1|0:1	0|0:0
2	105	.	AC	A	.	.	EXP=@121;BCSQ=@121	GT:BCSQ	1|0:1	0|0:0
2	121	.	AC	A	.	.	EXP=frameshift|ABC|ENST00000000002|-|11VVRTY>11*|105AC>A+121AC>A;BCSQ=frameshift|ABC|ENST00000000002|-|11VVRTY>11*|105AC>A+121AC>A	GT:BCSQ	1|0:1	0|0:0
2	126	.	C	CTT	.	.	EXP=@127;BCSQ=@127	GT:BCSQ	1|0:1	0|0:0
2	127	.	G	GG	.	.	EXP=inframe_insertion|ABC|ENST00000000002|-|9T>8TK|126C>CTT+127G>GG;BCSQ=inframe_insertion|ABC|ENST00000000002|-|9T>8TK|126C>CTT+127G>GG	GT:BCSQ	1|0:1	0|0:0
2	144	.	TAC	T	.	.	EXP=@148;BCSQ=@148	GT:BCSQ	1|0:1	0|0:0
2	148	.	TA	T	.	.	EXP=inframe_deletion&splice_region|ABC|ENST00000000002|-|5YV>5T|144TAC>T+148TA>T;BCSQ=inframe_deletion&splice_region|ABC|ENST00000000002|-|5YV>5T|144TAC>T+148TA>T	GT:BCSQ	1|0:1	0|0:0
2	164	.	T	G	.	.	EXP=missense|ABC|ENST00000000002|-|3T>3P|164T>G;BCSQ=missense|ABC|ENST00000000002|-|3T>3P|164T>G	GT:BCSQ	1|0:1	0|0:0
2	165	.	A	C	.	.	EXP=synonymous|ABC|ENST00000000002|-|2R>2R|165A>C;BCSQ=synonymous|ABC|ENST00000000002|-|2R>2R|165A>C	GT:BCSQ	1|0:1	0|0:0
2	169	.	A	G	.	.	EXP=@170;BCSQ=@170	GT:BCSQ	1|0:1	0|0:0
2	170	.	C	T	.	.	EXP=missense&start_lost|ABC|ENST00000000002|-|1V>1T|169A>G+170C>T;BCSQ=missense&start_lost|ABC|ENST00000000002|-|1V>1T|169A>G+170C>T	GT:BCSQ	1|0:1	0|0:0
2	199	.	G	T	.	.	EXP=5_prime_utr|ABC|ENST00000000002|-;BCSQ=5_prime_utr|ABC|ENST00000000002|-	GT:BCSQ	1|0:1	0|0:0
3	20	.	T	A	.	.	EXP=non_coding|mir-007|lincRNA;BCSQ=non_coding|mir-007|lincRNA	GT:BCSQ	1|0:1	0|0:0
3	109	.	ACGTACGT	A	.	.	EXP=splice_acceptor|QWRTY|ENST00000000003|-;BCSQ=splice_acceptor|QWRTY|ENST00000000003|-	GT:BCSQ	1|0:1	0|0:0
3	113	.	A	T	.	.	EXP=splice_region|QWRTY|ENST00000000003|-;BCSQ=splice_region|QWRTY|ENST00000000003|-	GT:BCSQ	1|0:1	0|0:0
3	120	.	T	A	.	.	EXP=intron|QWRTY;BCSQ=intron|QWRTY	GT:BCSQ	1|0:1	0|0:0
3	152	.	T	A	.	.	EXP=splice_region|QWRTY|ENST00000000003|-;BCSQ=splice_region|QWRTY|ENST00000000003|-	GT:BCSQ	1|0:1	0|0:0
3	159	.	G	A	.	.	EXP=splice_donor|QWRTY|ENST00000000003|-;BCSQ=splice_donor|QWRTY|ENST00000000003|-	GT:BCSQ	1|0:1	0|0:0
